prusti-common/src/vir/optimizations/mod.rs

```diff
@@ -15,25 +16,33 @@ pub mod predicates;
 
 impl Program {
     pub fn optimized(mut self) -> Self {
+        let optimizations = config::optimizations();
+        info!("Enabled optimisations: {:?}", optimizations);
+
         // can't borrow self because we need to move fields
-        let (new_methods, new_functions) =
-            functions::inline_constant_functions(self.methods, self.functions);
-        self.methods = new_methods
-            .into_iter()
-            .map(|m| {
-                let purified = methods::purify_vars(m);
-                folding::FoldingOptimizer::optimize(purified)
-            })
-            .collect();
-        self.functions = new_functions
-            .into_iter()
-            .map(|f| folding::FoldingOptimizer::optimize(f))
-            .collect();
-        self.viper_predicates = predicates::delete_unused_predicates(
-            &self.methods,
-            &self.functions,
-            self.viper_predicates,
-        );
+        if optimizations.contains(&Optimizations::All) || optimizations.contains(&Optimizations::InlineConstantFunctions) {
```

Comment:
I think it would be better to have something like `optimizations.is_inline_constant_functions_enabled()`.

Reply:
Since it seems that you need to do parsing manually anyway, you could probably have `Optimizations` as a struct with a boolean field for each optimization.